### PR TITLE
FIX: disconnect if socket closes

### PR DESCRIPTION
**Pull request by tacaswell**

### CODE REVIEW

File: caproto/threading/client.py [hunk 1]
```diff
@@ -391,8 +427,28 @@ def __init__(self, pvname, callback=None, form='time',
...
         # not handling lazy instantiation
         self._context.search(self.pvname)
+        self.chid = None
+        self.wait_for_connection(form=form, count=count)
+
+    @property
+    def connected(self):
+        return self.chid is not None and self.chid.connected
+
+    def force_connect(self, pvname=None, chid=None, conn=True, **kws):
```
Comment (danielballan) [inline caproto/threading/client.py]: I'm interested where you're going with this....

Comment (tacaswell) [inline caproto/threading/client.py]: that is a hold over from pyepics.

File: caproto/_state.py [hunk 2]
```diff
@@ -180,6 +181,7 @@
             ErrorResponse: MUST_CLOSE,
         },
         CLOSED: {
+            ClearChannelResponse: CLOSED,
```
Comment (danielballan) [inline caproto/_state.py]: This seems unnecessarily tolerant to me, especially because it's a *response*.

Comment (tacaswell) [inline caproto/_state.py]: thrashing to try to fix it ;)

Comment (tacaswell) [inline caproto/_state.py]: This is probably not the right fix anyway,  there seems to be a race condition when things are torn down where the channel is closed twice.

Comment (tacaswell) [inline caproto/_state.py]: and this copies the behavior in `FAILED` below.

**Comment by danielballan** [comment]
I forgot to submit that review from the train. I pushed a commit to address my own comments.

**Pull request merged**